### DISCUSSION TITLE
Which Controller?

### DISCUSSION THREAD

**Discussion by bs42** [zwave-js/zwave-js#1812] [Q&A]
I've got a HUSBZB-1 that is too slow.

I've now been through 2 of the ZST10-700 that when they work are nice and fast, but now both have failed after getting my network completely setup.

What is the most tested and functional controller for this?  What do the authors use?  I'm about to have to rebuild my zwave network again and I really want this next time to be the last time (at least for quite a while).

**Comment by AlCalzone** [comment]
I'm using a 500 series UZB for development and a 2016 Razberry hat for the production system. Both work just fine.

**Comment by varet80** [comment]
I have a HAT from z-wave.me Razberry 2016 for development (ex production) and a UZB-7 for production. they both seem to work nice
and possible to backup these using pc controller.

**Comment by scyto** [comment]
I don't recommend the razberry zwave pi hat  2018+ version based on me experience - its possible soldering an antenna on it might have fixed its capability, but it was a mess.

Happy with my SLUSB001A ~~once I increased gain by +4db~~ and firmware 7.15

**Comment by ahochsteger** [comment]
I'm using a [Z-Wave.me UZB (500 series)](https://z-wave.me/products/uzb/) for development and the same for production (with 60+ devices) it works quite well for me.
From time to time I do backups using Silicon Labs PC Controller and I've already successfully switched the dev and prod device by restoring the backups.

**Comment by jcam** [comment]
I've got a Zooz ZST10-700, it has been rock solid. Only issue I have is there's some sort of incompatibility with my host system so *if* i turn off the host, then after booting back up i have to unplug and plug back in the stick. Solution is to never power off the host system.

**Reply by jcam** [reply]
If I was going to buy another, I would buy the SLUSB001A

**Reply by bs42** [reply]
Both of mine have failed resulting this error: "ZWaveError: The controller is not yet ready!"

The first one I was told a firmware error had been located that may have been the cause and they sent out a replacement with updated firmware.  The second one failed today while trying to use it to update the firmware on a Zooz ZEN22 switch.

**Reply by AlCalzone** [reply]
Got any (complete) zwave-js logs?

**Reply by bs42** [reply]
It causes the server to fatal out with a backtrace, the value of debug in settings.json doesn't appear to impact the amount of output:

```
This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason:
ZWaveError: The controller is not yet ready!
    at Driver.get controller [as controller] (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:446:10)
    at BridgeApplicationCommandRequest.toLogEntry (/usr/src/app/node_modules/zwave-js/src/lib/controller/BridgeApplicationCommandRequest.ts:113:41)
    at DriverLogger.logMessage (/usr/src/app/node_modules/zwave-js/src/lib/log/Driver.ts:130:28)
    at Driver.serialport_onData (/usr/src/app/node_modules/zwave-js/src/lib/driver/Driver.ts:1412:19)
```

**Reply by AlCalzone** [reply]
That might already be enough. #1810
